codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttConnectPayload.java

```diff
@@ -37,6 +38,19 @@ public MqttConnectPayload(
             String password) {
         this.clientIdentifier = clientIdentifier;
         this.willTopic = willTopic;
+        this.willMessage = willMessage.getBytes(CharsetUtil.UTF_8);
+        this.userName = userName;
+        this.password = password.getBytes(CharsetUtil.UTF_8);
+    }
+
+     public MqttConnectPayload(
+            String clientIdentifier,
+            String willTopic,
+            byte[] willMessage,
+            String userName,
+            byte[] password) {
+        this.clientIdentifier = clientIdentifier;
+        this.willTopic = willTopic;
```

Comment:
I think we need to call `willTopic.clone()` and `password.clone()` here to ensure the user not later modify it.